core/node/api_server/src/execution_sandbox/vm_metrics.rs

```diff
@@ -132,16 +134,11 @@ pub(super) struct ExecutionMetrics {
 pub(super) static EXECUTION_METRICS: vise::Global<ExecutionMetrics> = vise::Global::new();
 
 pub(super) fn collect_tx_execution_metrics(
-    contracts_deployed: u16,
     result: &VmExecutionResultAndLogs,
 ) -> TransactionExecutionMetrics {
-    let writes_metrics = StorageWritesDeduplicator::apply_on_empty_state(&result.logs.storage_logs);
-    let event_topics = result
-        .logs
-        .events
-        .iter()
-        .map(|event| event.indexed_topics.len() as u16)
-        .sum();
+    let writes = StorageWritesDeduplicator::apply_on_empty_state(&result.logs.storage_logs);
+
+    // FIXME: these computations differ from ones in `VmExecutionResultAndLogs::get_execution_metrics()`; why?
     let l2_l1_long_messages = VmEvent::extract_long_l2_to_l1_messages(&result.logs.events)
```

Comment:
`VmExecutionResultAndLogs::get_execution_metrics` seems to be more precise. It shouldn't matter much since it's only used for debug/metrics

Reply:
I understand that it doesn't matter much, but the divergence itself looks weird. If it's possible to replace code here with `VmExecutionResultAndLogs::get_execution_metrics()`, will gladly do it since it'll make code simpler.